app/services/preserved_object_handler.rb

```diff
@@ -56,115 +28,110 @@ def initialize(druid, incoming_version, incoming_size, endpoint)
     if invalid?
-      results << result_hash(INVALID_ARGUMENTS, errors.full_messages)
+      handler_results.add_result(PreservedObjectHandlerResults::INVALID_ARGUMENTS, errors.full_messages)
     else
       Rails.logger.debug "update_version_after_validation #{druid} called"
       if endpoint.endpoint_type.endpoint_class == 'online'
         # NOTE: we deal with active record transactions in update_online_version, not here
         if moab_validation_errors.empty?
-          results.concat update_online_version(true, Status.ok)
+          update_online_version(true, Status.ok)
         else
-          results.concat update_online_version(true, Status.invalid_moab)
+          update_online_version(true, Status.invalid_moab)
         end
       elsif endpoint.endpoint_type.endpoint_class == 'archive'
         # TODO: perform archive object validation; then create a new PC record for the new
         #  archived version on the endpoint
       end
     end
 
-    log_results(results)
-    results
+    handler_results.log_my_results
+    handler_results.result_array
   end
 
   def update_version
-    results = []
     if invalid?
-      results << result_hash(INVALID_ARGUMENTS, errors.full_messages)
+      handler_results.add_result(PreservedObjectHandlerResults::INVALID_ARGUMENTS, errors.full_messages)
     else
       Rails.logger.debug "update_version #{druid} called"
       if endpoint.endpoint_type.endpoint_class == 'online'
         # NOTE: we deal with active record transactions in update_online_version, not here
-        results.concat update_online_version
+        update_online_version
       elsif endpoint.endpoint_type.endpoint_class == 'archive'
         # TODO: create a new PC record for the new archived version on the endpoint
       end
     end
 
-    log_results(results)
-    results
+    handler_results.log_my_results
+    handler_results.result_array
   end
 
   private
 
   def moab_validation_errors
-    results = []
     object_dir = "#{storage_location}/#{DruidTools::Druid.new(druid).tree.join('/')}"
     moab = Moab::StorageObject.new(druid, object_dir)
     object_validator = Stanford::StorageObjectValidator.new(moab)
-    errors = object_validator.validation_errors(Settings.moab.allow_content_subdirs)
-    if errors.any?
-      moab_error_msg_list = []
-      errors.each do |error_hash|
-        error_hash.each_value { |moab_error_msgs| moab_error_msg_list << moab_error_msgs }
+    moab_errors = object_validator.validation_errors(Settings.moab.allow_content_subdirs)
+    if moab_errors.any?
+      moab_error_msgs = []
+      moab_errors.each do |error_hash|
+        error_hash.each_value { |msg| moab_error_msgs << msg }
       end
-      results << result_hash(INVALID_MOAB, moab_error_msg_list)
-      log_results(results)
+      PreservedObjectHandlerResults.log_results(
+        [ handler_results.result_hash(PreservedObjectHandlerResults::INVALID_MOAB, moab_error_msgs) ]
+      )
     end
-    errors
+    moab_errors
```

Comment:
I refactored moab_validation_errors method in addition to the logging changes -- I think it's clearer now.  Happy to entertain dissent.